platform/ext/target/nordic_nrf/nrf5340/spm_hal.c

```diff
@@ -46,23 +62,53 @@ enum tfm_plat_err_t tfm_spm_hal_configure_default_isolation(
                   uint32_t partition_idx,
                   const struct tfm_spm_partition_platform_data_t *platform_data)
 {
-    /* FIXME we need to see how to handle this because nRF IDAU peripheral
-     * is not able to assign PRIV/nPRIV access policy configuration
+#if defined(CONFIG_TFM_ENABLE_MEMORY_PROTECT) && (TFM_LVL != 1)
+    bool privileged = tfm_is_partition_privileged(partition_idx);
+    struct mpu_armv8m_region_cfg_t region_cfg;
```

Comment:
Why not move this into the `if (!privileged)` block where it is only used? Actually, you could also not use the `privileged` variable and get rid of this `#if` here completely.

Reply:
I understood that this is because there are some "weird" coding style guidelines in tf-m that want us to define the variables in the top of the function

Reply:
I thought it was to be the top of a block, not necessarily a function. But if it is indeed about a function, you could always extract the block to be a separate function. Of course, it's not a big deal, I just find code that is less polluted with `#if`s a bit easier to read.

Reply:
No big deal either, I can go with your proposal (just wanted to communicate why I had done that this way)

Reply:
done now